### PR TITLE
ccxtbroker: Order fills, Cancel ID and more

### PR DESCRIPTION
**Pull request by Dave-Vallance**
This is the largest commit with the most amount of edits. 

NOTE: This some changes in this commit will need to work with the ccxtstore pull request coming soon.

The summary is

- Fix for the get_position method.
- Use the ccxt order ID when using the cancel_order method.
- Fix for passing params when calling create order
- Large change to track the Order filling status. This allow the position_size to be correctly set and also allows us to be notified when a stop loss order is triggered.
- Set startingcash and startingvalue during init. This allows us to plot charts after a run.

### CODE REVIEW

File: backtrader/brokers/ccxtbroker.py [hunk 1]
```diff
@@ -71,15 +79,37 @@ def get_notification(self):
...
+        if clone:
+            pos = pos.clone()
+
+        return pos
+
+    def next(self):
+        for o_order in list(self.open_orders):
+            oID = o_order.ccxt_order['id']
+            ccxt_order = self.store.fetch_order(oID)
+            if ccxt_order['status'] == 'closed':
```
Comment (bartosh) [inline backtrader/brokers/ccxtbroker.py:94]: ccxt specifics should go to store. broker and feed shouldn't contain them.

File: backtrader/brokers/ccxtbroker.py [hunk 2]
```diff
@@ -71,15 +79,37 @@ def get_notification(self):
...
+                pos.update(o_order.size, o_order.price)
+                o_order.completed()
+                self.notify(o_order)
+                self.open_orders.remove(o_order)
 
     def _submit(self, owner, data, exectype, side, amount, price, params):
-        order_type = self.order_types.get(exectype)
+        order_type = self.order_types.get(exectype) if exectype else 'market'
+        # Extract CCXT specific params if passed to the order
+        params = params['params'] if 'params' in params else params
```
Comment (bartosh) [inline backtrader/brokers/ccxtbroker.py]: Would this be a bit more pythonic?
```
params = params.get('params', params)
```

**Review by bartosh — COMMENTED** [review]
I'm afraid I don't fully understand this PR. Can you split it to several PRs and comment them in a bit more detailed way?

File: backtrader/brokers/ccxtbroker.py (hunk 1 above)
Comment (Dave-Vallance) [reply backtrader/brokers/ccxtbroker.py:94]: Hi Ed, 
  Isn't the CCXTbroker by definition CCXT specific? I guess it is a matter of preference. I placed it in the broker as I saw it as a broker type operation. I.e tracking whether our order was filled and giving the appropriate notification.

File: backtrader/brokers/ccxtbroker.py (hunk 2 above)
Comment (Dave-Vallance) [reply backtrader/brokers/ccxtbroker.py]: Sure! :) That looks better!

File: backtrader/brokers/ccxtbroker.py [hunk 3]
```diff
@@ -71,15 +79,37 @@ def get_notification(self):
...
-    def getposition(self, data):
-        currency = data.symbol.split('/')[0]
-        return self.store.getposition(currency)
+    def getposition(self, data, clone=True):
+        # return self.o.getposition(data._dataname, clone=clone)
+        pos = self.positions[data._dataname]
+        if clone:
+            pos = pos.clone()
+
+        return pos
```
Comment (FiV0) [inline backtrader/brokers/ccxtbroker.py:88]: This doesn't work if the you start the strategy when you are already holding a position.

File: backtrader/brokers/ccxtbroker.py [hunk 4]
```diff
@@ -96,7 +126,19 @@ def sell(self, owner, data, size, price=None, plimit=None,
...
-        return self.store.cancel_order(order['id'])
+        oID = order.ccxt_order['id']
+        # check first if the order has already been filled otherwise an error
+        # might be raised if we try to cancel an order that is not open.
+        ccxt_order = self.store.fetch_order(oID)
+        if ccxt_order['status'] == 'closed':
+            return order
+
+        ccxt_order = self.store.cancel_order(oID)
+        if ccxt_order['status'] == 'canceled':
```
Comment (FiV0) [inline backtrader/brokers/ccxtbroker.py:137]: In case of binance this gives an `key error` as the canceled order no longer contains a `status` field.

**Pull request closed without merging**